### ISSUE TITLE
Recent update of `rustup-init` to `rustup` broke Rust-related commands (cargo, rustc, etc)

### ISSUE THREAD

**Issue by rben01** [Homebrew/homebrew-core#178311]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
$ brew config
HOMEBREW_VERSION: 4.3.10
ORIGIN: https://github.com/Homebrew/brew
HEAD: c3ae0327c01aab461531bfc0fcdfb4dd3a779205
Last commit: 6 days ago
Core tap JSON: 24 Jul 14:30 UTC
Core cask tap JSON: 24 Jul 14:30 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.3 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.3/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.0.1
Rosetta 2: false
```

### Verification

- [X] My `brew doctor` output says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [ ] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

`brew doctor` output

```
Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  cowsay
  youtube-dl

Warning: Your Xcode (15.0.1) is outdated.
Please update to Xcode 15.4 (or delete it).
Xcode can be updated from the App Store.
```

### What were you trying to do (and why)?

After running `brew update`, formula `rustup-init` was moved to `rustup`. After this change, Rust-related commands such as `cargo` and `rustc` stopped working; my shell considered them nonexistent. It appears that the symlinks in `$HOME/.cargo/bin/cargo` still existed, but were broken (where they pointed didn't exist anymore).

### What happened (include all command output)?

`$ cargo run --release`

`fish: Unknown command: cargo`

`$ brew reinstall rustup` fixed the issue

### What did you expect to happen?

Commands installed through brew `rustup` should continue to exist.

### Step-by-step reproduction instructions (by running `brew` commands)

`$ brew update`
This moved `rustup-init` to `rustup`, and in the process seems to have broken Rust-related commands

The fix was to `brew reinstall rustup`, but this should not be necessary.

Unfortunately I cannot reproduce this issue because I cannot go back in time to when `rustup-init` was installed.

**Comment by carlocab** [comment]
Sounds like breakage in formula renames. CC @Homebrew/brew for thoughts.

**Comment by branchvincent** [comment]
I created #178322 before I saw this issue based on https://github.com/Homebrew/homebrew-core/pull/177840#issuecomment-2248195278, but it's from the rename logic moving the formulae to keg-only but not running: https://github.com/Homebrew/homebrew-core/blob/4d82c6773e77c809ca47b35c0aeca9b7abfdcd5c/Formula/r/rustup.rb#L41-L43

**Comment by bennesp** [comment]
*Workaround*: `brew uninstall rustup && brew install rustup` solved the issue for me

**Comment by p-linnane** [comment]
This is fixed after #178322. Running `brew upgrade rustup` is all that's required.
